mp/src/game/server/momentum/mom_lobby_system.cpp

```diff
@@ -650,18 +655,49 @@ void CMomentumLobbySystem::HandlePersonaCallback(PersonaStateChange_t* pParam)
         return;
 
     const auto person = CSteamID(pParam->m_ulSteamID);
+    const char *pName = SteamFriends()->GetFriendPersonaName(person);
+
     if (pParam->m_nChangeFlags & k_EPersonaChangeName)
     {
         if (IsInLobby(person))
         {
             const auto pGhost = GetLobbyMemberEntity(pParam->m_ulSteamID);
             if (pGhost)
             {
-                const char *pName = SteamFriends()->GetFriendPersonaName(person);
                 pGhost->SetGhostName(pName);
             }
         }
     }
+
+    if (pParam->m_nChangeFlags & k_EPersonaChangeRelationshipChanged)
+    {
+        if (MomUtil::IsSteamUserBlocked(person.ConvertToUint64()))
+        {
+            Warning("%s was just blocked!\n", pName);
+
+            if (!m_vecBlocked.HasElement(person))
+            {
+                m_vecBlocked.AddToTail(person);
+            }
+
+            const auto findIndex = m_mapLobbyGhosts.Find(person.ConvertToUint64());
+
+            if (m_mapLobbyGhosts.IsValidIndex(findIndex))
+            {
+                // This player was just blocked by us and their ghost currently exists, remove it
+                GetLobbyMemberEntity(person)->Remove();
```

Comment:
Could you also try this? Could save doing two finds.
```suggestion
                m_mapLobbyGhosts[findIndex]->Remove();
```